### ISSUE TITLE
Treemacs causes Emacs to hang/freeze

### ISSUE THREAD

**Issue by knl** [syl20bnr/spacemacs#12249]
#### Description :octocat:
Treemacs causes Emacs to hang/freeze

#### Reproduction guide :beetle:
* Start Emacs
* Open any file within the projectile
* Open Treemacs with `SPC p t`
* Keep working with files, opening them by clicking on Treemacs' sidebar, editing, ...
* Be patient

*Observed behaviour:* :eyes: :broken_heart:
- Note that the whole Emacs freezes and becomes unresponsive to any input, including ESC/C-g

*Expected behaviour:* :heart: :smile:
- No freezes/hangs

#### System Info :computer:
- OS: darwin
- Emacs: 27.0.50
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 3d11456f5)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm auto-completion emacs-lisp git markdown
      (treemacs :variables treemacs-use-follow-mode t)
      org
      (shell :variables shell-default-height 30 shell-default-position 'bottom shell-default-term-shell "/usr/local/bin/zsh")
      (spell-checking :variables enable-flyspell-auto-completion t)
      spell-checking syntax-checking version-control haskell
      (python :variables python-test-runner 'pytest)
      (javascript :variables javascript-disable-tern-port-files nil)
      lua go html yaml terraform nixos emacs-lisp shell-scripts javascript perl5 themes-megapack)
```
- System configuration features: RSVG IMAGEMAGICK GLIB NOTIFY KQUEUE ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS JSON PDUMPER LCMS2 GMP

#### Backtrace :paw_prints:
Can't provide any, as I have to kill the whole Emacs process.

**Comment by Compro-Prasad** [comment]
I have a similar issue with different configuration. Can't get it on spacemacs as I have disabled many things that use CPU. See if you can dig it further.

https://github.com/Alexander-Miller/treemacs/issues/413

**Comment by Alexander-Miller** [comment]
> Note that the whole Emacs freezes and becomes unresponsive to any input, including ESC/C-g

What about `kill -12`?

**Comment by gcentauri** [comment]
I've had some issues with this too, my suspicion is that it is related to interactions with version control / `magit`.  I don't have any proof, but I'll try to gather any data if I notice it happening again.

**Comment by Alexander-Miller** [comment]
Good point. I remember there being a potential endless loop in treemacs-magit that I fixed a while ago, so best give it an update if you use it.
And update treemacs as well. With a bit of luck https://github.com/Alexander-Miller/treemacs/commit/d83d33ac78f94f8b91df2ff06ce79e55f2d6b66b will have been the fix.

**Comment by knl** [comment]
@Alexander-Miller seems the issue is not treemacs, as even with treemacs off, my Emacs hangs when editing terraform and Nix files. So, let's change the title of the issue, or merge with another. I'm looking to narrow down the problem. `kill -12` kills the app. If I run with `SPC t D` on, I get too many faults in `company` for `terraform` files.

**Comment by Alexander-Miller** [comment]
> as even with treemacs off, my Emacs hangs when editing terraform and Nix files

Off as in not visible or not even loaded/used? If a treemacs exists it'll still react to your editing files if those files are in a treemacs project.

**Comment by knl** [comment]
Sorry for radio silence, work-overload, and I wanted to make sure I did everything correctly. After doing several days of work with treemacs completely off *and* never invoked (even preventing loading), I noticed I still had hangs. Then, by pure chance, I noticed that aspell process being active -- once I uninstalled aspell from my machine, there are no hangs anymore. So, maybe this issue should be closed, or renamed.

Alexander, thanks for treemacs btw, I use it daily, and helps me navigate hierarchies easily.
